src/AtomicLaunch.sol

```diff
@@ -0,0 +1,281 @@
+            "<pool-curve-citadelToLiquidity!"
+        );
+        require(
+            pool.balances(1) >= wbtcToLiquidity,
+            "<pool-curve-wbtcToLiquidity!"
+        );
+
+        _deployOracles(poolAddress);
+
+        _fundingContractsInitialize();
+
+        // NOTE: mainly for clean event, so no needs for mega-detective work
+        emit PoolCreated(poolAddress);
+    }
+
+    function _initialPoolPrice() internal returns (uint256 _poolPrice) {
+        (uint256 wbtcOraclePricing, , bool valid) = safeLatestAnswer(
+            wbtcUsdPriceFeed
+        );
+
+        require(valid, "not-valid!");
+
+        uint256 wbtcPrice = wbtcOraclePricing / 10**wbtcUsdPriceFeed.decimals();
+
+        _poolPrice = (wbtcPrice / CITADEL_PRICE) * 1e18;
+    }
+
+    function _deployOracles(address _ctdlWbtcCurvePool) internal {
+        CtdlWbtcCurveV2Provider ctdlWbtcProviderLoc = new CtdlWbtcCurveV2Provider(
+                _ctdlWbtcCurvePool
+            );
+        _oracles.add(address(ctdlWbtcProviderLoc));
+
+        CtdlBtcChainlinkProvider ctdlBtcProvider = new CtdlBtcChainlinkProvider(
+            _ctdlWbtcCurvePool,
+            WBTC_BTC_PRICE_FEED
+        );
+        _oracles.add(address(ctdlBtcProvider));
+
+        CtdlWibbtcLpVaultProvider ctdlWibbtcProvider = new CtdlWibbtcLpVaultProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                WIBBTC_LP_VAULT
+            );
+        _oracles.add(address(ctdlWibbtcProvider));
+
+        CtdlEthChainlinkProvider ctdlEthProvider1 = new CtdlEthChainlinkProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                BTC_ETH_PRICE_FEED
+            );
+        _oracles.add(address(ctdlEthProvider1));
+
+        CtdlAssetChainlinkProvider ctdlEthProvider2 = new CtdlAssetChainlinkProvider(
+                _ctdlWbtcCurvePool,
+                WBTC_BTC_PRICE_FEED,
+                address(wbtcUsdPriceFeed),
+                ETH_USD_PRICE_FEED
+            );
+        _oracles.add(address(ctdlEthProvider2));
+
+        address[4] memory assetEthFeeds = [
+            FRAX_ETH_PRICE_FEED,
+            USDC_ETH_PRICE_FEED,
+            CVX_ETH_PRICE_FEED,
+            BADGER_ETH_PRICE_FEED
+        ];
+
+        address[4] memory assetUsdFeeds = [
+            FRAX_USD_PRICE_FEED,
+            USDC_USD_PRICE_FEED,
+            CVX_USD_PRICE_FEED,
+            BADGER_USD_PRICE_FEED
+        ];
+
+        for (uint256 i; i < 4; ++i) {
+            CtdlAssetChainlinkProvider ctdlAssetProvider1 = new CtdlAssetChainlinkProvider(
+                    _ctdlWbtcCurvePool,
+                    WBTC_BTC_PRICE_FEED,
+                    BTC_ETH_PRICE_FEED,
+                    assetUsdFeeds[i]
+                );
+            _oracles.add(address(ctdlAssetProvider1));
+
+            CtdlAssetChainlinkProvider ctdlAssetProvider2 = new CtdlAssetChainlinkProvider(
+                    CTDL_WBTC_CURVE_POOL,
```

Comment:
CTDL_WBTC_CURVE_POOL is not defined, shouldn't it be _ctdlWbtcCurvePool?

Reply:
should be resolved in latest commit